self-core-impl/src/main/java/com/selfxdsd/core/GitlabCommits.java

```diff
@@ -0,0 +1,159 @@
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.util.Iterator;
+
+/**
+ * Commits in a Gitlab Repo.
+ * @author Ali Fellahi (fellahi.ali@gmail.com)
+ * @version $Id$
+ * @since 0.0.44
+ */
+final class GitlabCommits implements Commits {
+
+    /**
+     * Logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(
+        GitlabCommits.class
+    );
+
+    /**
+     * Gitlab repo Commits base uri.
+     */
+    private final URI commitsUri;
+
+    /**
+     * Gitlab's JSON Resources.
+     */
+    private final JsonResources resources;
+
+    /**
+     * Self storage, in case we want to store something.
+     */
+    private final Storage storage;
+
+    /**
+     * Ctor.
+     *
+     * @param resources Gitlab's JSON Resources.
+     * @param commitsUri Commits base URI.
+     * @param storage Storage.
+     */
+    GitlabCommits(
+        final JsonResources resources,
+        final URI commitsUri,
+        final Storage storage
+    ) {
+        this.resources = resources;
+        this.commitsUri = commitsUri;
+        this.storage = storage;
+    }
+
+    @Override
+    public Commit getCommit(final String ref) {
+        LOG.debug(
+            "Getting commit [" + ref + "] from ["
+            + this.commitsUri + "..."
+        );
+        final URI commitUri = URI.create(
+            this.commitsUri.toString() + "/" + ref
+        );
+        final Resource resource = this.resources.get(commitUri);
+        final JsonObject jsonObject;
+        switch (resource.statusCode()) {
+            case HttpURLConnection.HTTP_OK:
+                jsonObject = resource.asJsonObject();
+                break;
+            case HttpURLConnection.HTTP_NOT_FOUND:
+                jsonObject = null;
+                break;
+            default:
+                LOG.error(
+                    "Could not get the commit " + ref + ". "
+                    + "Received status code: " + resource.statusCode() + "."
+                );
+                throw new IllegalStateException(
+                    "Could not get the commit " + ref + ". "
+                    + "Received status code: " + resource.statusCode() + "."
+                );
+        }
+        final Commit commit;
+        if(jsonObject != null){
+            LOG.debug(String.format("Commit [%s] found!", ref));
+            commit = new GitlabCommit(
+                commitUri,
+                jsonObject,
+                this.storage,
+                this.resources
+            );
+        } else {
+            LOG.debug("Commit [" + ref + "] not found, returning null.");
+            commit = null;
+        }
+        return commit;
+    }
+
+    @Override
+    public Commit latest() {
+        final Resource resource = this.resources.get(this.commitsUri);
+        final Commit latest;
+        if (resource.statusCode() == HttpURLConnection.HTTP_OK) {
+            final JsonObject json = resource.asJsonArray().getJsonObject(0);
```

Comment:
@amihaiemil Here I followed the implementation from `GithubCommits` 
but I think we should check  `resource.asJsonArray()` is not empty before calling `.getJsonObject(0)` just in case?

Reply:
@fellahi-ali well, there will always be at least one commit in the Github Repo. 

The Commits are used when handling PUSH event (to find the todos and leave a puzzle). So if we have a push event, it means there is at least 1 commit in the repo. 

So I guess it's fine.